cmd/acr-credential-provider/main.go

```diff
@@ -68,6 +75,8 @@ func main() {
 	// Flags
 	command.Flags().StringVarP(&RegistryMirrorStr, "registry-mirror", "r", "",
 		"Mirror a source registry host to a target registry host, and image pull credential will be requested to the target registry host when the image is from source registry host")
+	command.Flags().StringToStringVar(&KSAAuthConfigMap, "ksa-auth-config", map[string]string{},
+		"KSA auth configuration as key=value pairs (sni-name=xxx,default-client-id=xxx,default-tenant-id=xxx,api-server-ip=xxx)")
```

Comment:
I think this is not very “standard" and structural? Generally we should have 1 parameter for one configuration? Otherwise you can use one parameter to pass everything.

Reply:
This is a well-established approach for popular CLI tools when applying grouped configurations or options, such as:
```bash
docker buildx create \
  --driver-opt network=host \
  --driver-opt image=moby/buildkit:latest
```
or
```
docker run --mount type=bind,source=/src,target=/dst,readonly
```

Reply:
To simplify input, we might be able to do something like `--ksa-auth sni=xxx,ip=xxx,default-client=xxx`.

Reply:
1. For Docker, I think they do that because the CLI is just a proxy to the Docker daemon, and this approach preserves compatibility.
2. Are they optional?
3. Separate flags could provide more helpful descriptions for each one, as well as clearer type restrictions.